### ISSUE TITLE
Can not build small library after upgrade rules to 6.1

### ISSUE THREAD

**Issue by farcop** [bazelbuild/rules_jvm_external#1177]
```
# MODULE.bazel
bazel_dep(name = "rules_java", version = "7.6.4")
bazel_dep(name = "rules_jvm_external", version = "6.1")

maven = use_extension("@rules_jvm_external//:extensions.bzl", "maven")

version_keycloak = "23.0.7"

maven.install(
    artifacts = [
        "jakarta.ws.rs:jakarta.ws.rs-api:3.1.0",
    ],
    fail_if_repin_required = True,
    lock_file = "//:maven_install.json",
)

[
    maven.artifact(
        artifact = artifact,
        group = "org.keycloak",
        neverlink = True,
        version = version_keycloak,
    )
    for artifact in [
        "keycloak-core",
        "keycloak-model-jpa",
        "keycloak-server-spi",
        "keycloak-server-spi-private",
    ]
]

use_repo(maven, "maven", "unpinned_maven")
```

```
# BUILD
load("@rules_java//java:defs.bzl", "java_library")

java_library(
    name = "jar",
    srcs = glob(["src/main/java/org/keycloak/kublr_extension/**/*.java"]),
    resources = glob(["src/main/resources/META-INF/**/*"]),
    deps = [
        "@maven//:jakarta_ws_rs_jakarta_ws_rs_api",
        "@maven//:org_keycloak_keycloak_core",
        "@maven//:org_keycloak_keycloak_model_jpa",
        "@maven//:org_keycloak_keycloak_server_spi",
        "@maven//:org_keycloak_keycloak_server_spi_private",
    ],
)
```

After upgrade to 6.1 I updated MODULE.bazel.lock successfully
`bazel mod deps --lockfile_mode=update`

But when I try to build I got error...
```
bazel build //images/keycloak/kublr-extension:jar
ERROR: Analysis of target '//images/keycloak/kublr-extension:jar' failed; build aborted: MODULE.bazel.lock is no longer up-to-date because: The repo mappings of certain repos used by the extension 'ModuleExtensionId{bzlFileLabel=@@rules_jvm_external~//:extensions.bzl, extensionName=maven, isolationKey=Optional.empty}' have changed. Please run `bazel mod deps --lockfile_mode=update` to update your lockfile.
INFO: Elapsed time: 0.153s, Critical Path: 0.04s
INFO: 1 process: 1 internal.
ERROR: Build did NOT complete successfully
```

This project successfully builds with rules version 6.0

**Comment by farcop** [comment]
Can confirm fix. Just test with single_version_override() with [patch](https://patch-diff.githubusercontent.com/raw/bazelbuild/rules_jvm_external/pull/1171.patch)
Waiting for release...

**Comment by shs96c** [comment]
We have now shipped `rules_jvm_external` 6.2, which should have this fix in it. Closing this, but please file another issue if the problem reoccurs.
